docs/getting_started.rst

```diff
@@ -0,0 +1,394 @@
+about compiling their own functions.  Using Xarray functions and the boolean mask is
+generally enough to accomplish most tasks, and this mask is often more efficient to operate
+on due to its high on-disk compression ratio.  These examples better demonstrate the distinction between
+the two as well as ways in which they are commonly used:
+
+.. ipython:: python
+
+    import sgkit as sg
+    ds = sg.simulate_genotype_call_dataset(n_variant=1, n_sample=4, n_ploidy=2, missing_pct=.3, seed=4)
+    ds.call_genotype
+
+    # Count alternate alleles while omitting partial calls
+    ##############
+    # Using Xarray
+    ##############
+    import xarray as xr
+    alt_allele_count = xr.where(
+        # Identify where there are any missing calls across chromosomes
+        ds.call_genotype_mask.any(dim='ploidy'),
+        -1, # Return -1 if any one call for a chromosome is missing
+        (ds.call_genotype > 0).sum(dim='ploidy') # Otherwise, sum non-ref calls
+    )
+    # Note that only the first two samples have meaningful counts since
+    # at least one call is missing for the last two samples
+    alt_allele_count.values
+
+    #############
+    # Using Numba
+    #############
+    import numba
+    import numpy as np
+
+    def alt_allele_count(gt):
+        out = np.full(gt.shape[:2], -1, dtype=np.int64)
+        for i, j in np.ndindex(*out.shape):
+            if np.all(gt[i, j] >= 0):
+                out[i, j] = np.sum(gt[i, j] > 0)
+        return out
+
+    # Jit-compiled functions are often simpler with a single array input, since
+    # conditional logic based on sentinel values is easier to program with this API
+    numba.njit(alt_allele_count)(ds.call_genotype.values)
+
+Again, this is not necessarily a level of detail most users should need to worry about. It is important
+to understand though when constructing more complex workflows that take full advantage of the many
+tools in the `PyData <https://pydata.org/>`_ ecosystem.
+
+
+Chaining operations
+~~~~~~~~~~~~~~~~~~~
+
+`Method chaining <https://tomaugspurger.github.io/method-chaining.html>`_ is a common practice with Python
+data tools that improves code readability and reduces the probability of introducing accidental namespace collisions.
+Sgkit functions are compatible with this idiom by default and this example shows to use it in conjunction with
+Xarray and Xandas operations in a single pipeline:
+
+.. ipython:: python
+
+    import sgkit as sg
+    ds = sg.simulate_genotype_call_dataset(n_variant=100, n_sample=50, missing_pct=.1)
+
+    # Use `pipe` to apply a single sgkit function to a dataset
+    ds_qc = ds.pipe(sg.variant_stats).drop_dims('samples')
+    ds_qc
+
+    # Show statistics for one of the arrays to be used as a filter
+    ds_qc.variant_call_rate.to_series().describe()
+
+    # Build a pipeline that filters on call rate and computes Fst between two populations
```

Comment:
Maybe add "(or cohorts)" to echo the usage below.

Reply:
https://github.com/pystatgen/sgkit/commit/f7f5a039412fc6b4c29621d283c1a5738b746521#diff-9bc8bf6e8d9db020cefce1fbc0426795R268